crud/select/plan.lua

```diff
@@ -235,6 +277,17 @@ function select_plan.new(space, conditions, opts)
         sharding_key = extract_sharding_key_from_scan_value(scan_value, scan_index, sharding_index)
     end
 
+    if sharding_key == nil then
+        -- Ignore possible errors to preserve old behavior
+        -- since here it affects only extracting sharding_key from
+        -- conditions and it should not be critical to select call.
+        local ddl_sharding_key = sharding_key_module.fetch_on_router(space_name)
```

Comment:
Let's use `opts.sharding_key_as_index_obj`.

BTW, I vote for keeping naming of this object the same. Since PR #181 introduces `sharding_key_as_index_obj`, let's continue.

Reply:
Reworked to use with existing values, keep naming in new version.